historical-adherence-query/java/Main.java

```diff
@@ -0,0 +1,203 @@
+            }
+        }
+    }
+
+    public class HistoricalAdherenceEventListener implements
+        NotificationListener<WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice> {
+        private String topic;
+        private HashMap<String, WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice> resultMap = new HashMap<>();
+
+        public Class<WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice> getEventBodyClass() {
+            return WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice.class;
+        }
+
+        public String getTopic() {
+            return topic;
+        }
+
+        public HistoricalAdherenceEventListener(String userId) {
+            // This is the topic to which the notification handler subscribes
+            this.topic = "v2.users." + userId + ".workforcemanagement.historicaladherencequery";
+        }
+
+        // This allows you to retrieve the result after it has been completed
+        public WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice getResultForId(String operationId) {
+            return resultMap.get(operationId);
+        }
+
+        // This is used to check if the request has been completed for a given operations ID
+        public Boolean isCompleted(String operationId){
+            if (resultMap.containsKey(operationId)) {
+                WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice result = resultMap.get(operationId);
+                if (result.getQueryState() == WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice.QueryStateEnum.COMPLETE) {
+                    //the request has been completed
+                    return true;
+                }
+                // If the result is not completed, you can determine how to handle that, here we will just throw an exception with the unexpected state
+                throw new NoSuchElementException("The resulting state for " + operationId + " was " + result.getQueryState());
+            }
+            // The request has not yet been completed
+            return false;
+        }
+
+        // This is the callback for historical adherence events, called when a query is made.
+        @Override
+        public void onEvent(NotificationEvent<?> event) {
+            // Extract the WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice object from the eventBody
+            WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice result = (WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice) event.getEventBody();
+            resultMap.put(result.getId(), result);
+            // Get the necessary details
```

Comment:
What are the necessary details and what would you do with them? Either show some kind of usage here or explain how someone would make use of this in the tutorial.